src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1594,6 +1606,10 @@ def status(self, id):
 
     @private
     def close_libvirt_connection(self):
+        if self.libvirt_event_loop_thread:
+            self.libvirt_event_loop_thread_condition = False
+            self.libvirt_event_loop_thread.join(2.0)
```

Comment:
I am not sure why you need to join() then? Wont the thread only end after you close the connection in the next lines, because it will likely be blocked within the loop?